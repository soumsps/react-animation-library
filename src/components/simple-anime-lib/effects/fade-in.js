import React, { useRef, memo } from 'react';
import { DEFAULT_FADEIN_CONFIG } from '../constants';
import {
  useAnimationFrame,
  timingFunction,
  getDelay,
  getDuration,
  getIterationCount,
  getEasing,
  getCallback,
} from '../helper';

const FadeIn = memo((props) => {
  const delay = getDelay(props.delay, DEFAULT_FADEIN_CONFIG.delay);
  const duration = getDuration(props.duration, DEFAULT_FADEIN_CONFIG.duration);
  const iterationCount = getIterationCount(
    props.iterationCount,
    DEFAULT_FADEIN_CONFIG.iterationCount
  );
  const easing = getEasing(props.easing, DEFAULT_FADEIN_CONFIG.easing);
  const callback = getCallback(props.callback, DEFAULT_FADEIN_CONFIG.callback);
  const timing = timingFunction[easing];
  const elementRef = useRef({ style: { opacity: 0.0 } });

  const draw = (progress) => {
    if (elementRef.current !== null) {
      elementRef.current.style.opacity = progress;
    }
  };

  useAnimationFrame({
    delay,
    duration,
    callback,
    iterationCount,
    timing,
    draw,
  });
  return (
    <span ref={elementRef} style={elementRef.current.style}>
      {props.children}
    </span>
  );
});

export { FadeIn };
