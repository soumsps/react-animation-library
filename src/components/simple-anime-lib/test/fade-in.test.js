import React from 'react';
import renderer from 'react-test-renderer';
import { FadeIn } from '../fade-in';

it('fade-in effect snapshot test', () => {
  const tree = renderer.create(<FadeIn></FadeIn>).toJSON();
  expect(tree).toMatchSnapshot();
});
