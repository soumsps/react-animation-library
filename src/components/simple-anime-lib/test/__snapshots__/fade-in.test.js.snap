// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fade-in effect snapshot test 1`] = `
<span
  style={
    Object {
      "opacity": 0,
    }
  }
/>
`;
